src/validation.ts

```diff
@@ -1,43 +1,35 @@
-import { 
-  PayloadType,
-  ContractCallPayload,
-  getAbi,
-  addressToString,
-  StacksTransaction
-} from "@stacks/transactions";
+import { PayloadType, ContractCallPayload, getAbi, addressToString, StacksTransaction } from '@stacks/transactions';
 
 // rules for transaction sponsorship eligibility
 // customize this if you are forking this repo
 export async function validateTransaction(transaction: StacksTransaction) {
   if (transaction.payload.payloadType !== PayloadType.ContractCall) {
     throw new Error('Transaction is not a contract call');
-  } 
+  }
 
   const payload = transaction.payload as ContractCallPayload;
 
   // check and limit to SIP-09 transfers calls
-  if (payload.functionName.content.toString() !== 'transfer') {
-    throw new Error('Transaction is not a NFT transfer contract call');
-  } 
+  if (payload.contractAddress.hash160.toString() !== 'e685b016b3b6cd9ebf35f38e5ae29392e2acd51d') {
+    throw new Error('Transaction is not an ALEX swap contract call');
+  }
 
   const contractAddress = addressToString(payload.contractAddress);
   const contractName = payload.contractName.content.toString();
   const functionName = payload.functionName.content.toString();
-  const abi = await getAbi(contractAddress, contractName, "mainnet");
-  const func = abi.functions.find((func) => {
-    return func.name === functionName;
-  })
+  const abi = await getAbi(contractAddress, contractName, 'mainnet');
+  const func = abi.functions.find((fn) => {
+    return fn.name === functionName;
+  });
 
   if (func.args.length !== 3) {
     throw new Error('Transaction is not a NFT transfer contract call');
   }
 
   // check against sip-09 interface for transfer
-  if (func.args[0].type !== 'uint128'
-    || func.args[1].type !== 'principal'
-    || func.args[2].type !== 'principal') {
+  if (func.args[0].type !== 'uint128' || func.args[1].type !== 'principal' || func.args[2].type !== 'principal') {
```

Comment:
the check for argument type needs to be removed as this is for SIP9 NFT transfers